### DISCUSSION TITLE
All buttons are named "Fridge Freezer"

### DISCUSSION THREAD

**Discussion by V1ru8** [thoukydides/homebridge-homeconnect#102] [Q&A] [answered]
![IMG_A03098F6C7E2-1](https://user-images.githubusercontent.com/473778/189727390-79499184-9392-4dff-a5a9-eef14d7857ab.jpeg)
All buttons of my fridge/freezer are named "Fridge Freezer" in HomeKit. It's impossible to figure out what is what.
One home bridge, it looks like they have better names:
<img width="1529" alt="Screen Shot 2022-09-12 at 11 19 38 AM" src="https://user-images.githubusercontent.com/473778/189727607-b802b55d-4881-454b-94a3-6fff8b546023.png">
Is there a way to make these names show up in HomeKit?

Also I'm missing a button to enable/disable my ice maker. It doesn't seem to show up. Any logging information I can provide to see whi it's not showing up? Or is it not supported?

**Comment by thoukydides** [comment, marked as answer]
> All buttons of my fridge/freezer are named "Fridge Freezer" in HomeKit. It's impossible to figure out what is what.

Yes, that's a change that Apple made in the iOS 16 version of their Home app. It now sets the names of all Services to that of the parent Accessory. The name provided by the Homebridge plugin for each Service now only used as a placeholder when editing the name. See https://github.com/homebridge/homebridge/issues/3210.

Unfortunately, this also affects control via Siri; it isn't just a display issue within the Home app itself.

> One home bridge, it looks like they have better names:

Those names are also shown by the iOS 15 (and earlier) versions of the Home app, and by third-party HomeKit apps.

> Is there a way to make these names show up in HomeKit?

Yes. You can edit the names within the Home app. If you open the settings for each switch and delete the existing name then it will show the name provided by the plugin as a placeholder. You can then set your own name. Unfortunately, you need to manually repeat this process for each switch.

The only way that I could change the behaviour from the plugin would be to expose each Service (switch) as a separate Accessory. However, they would then appear completely unrelated in the Home app and third-party HomeKit apps, so I don't think that's a desirable change.

Apple are planning to release an "Updated architecture" for HomeKit later this year (presumably next month after iPadOS has also been updated to version 16). It is quite likely that this will introduce further changes, so it doesn't make sense to rush into changing anything now.

I recommend submitting feedback to Apple at: https://www.apple.com/feedback/iphone/

> Also I'm missing a button to enable/disable my ice maker. It doesn't seem to show up. Any logging information I can provide to see whi it's not showing up? Or is it not supported?

As I explained in #94 the Home Connect API does not currently support ice makers.

I recommend requesting this feature at: https://developer.home-connect.com/support/contact

**Reply by V1ru8** [reply]
Thank you so much for the detailed reaply!

**Comment by thoukydides** [comment]
Based on a [Discord](https://discord.com/channels/432663330281226270/432672072859385856/1040394094528188416) discussion I have made some changes in v0.27.0 that help. (See #116 for more details.)
